### ISSUE TITLE
Question regarding the use of T5 with the german language

### ISSUE THREAD

**Issue by TiloMichel** [d-e-h-i-o/german-qg#1]
Hey, I found your models at hugging face and was wondering why you used the fine-tuned T5 model to fine-tune it again on the question generation task.
I read the T5 paper and the C4 dataset to pre-train T5 only contains English language as well as the SQuAD dataset which is used for the question generation fine-tuning.
I think it would have made more sense to fine-tune a mT5 model since it already "learned" some german.

**Comment by d-e-h-i-o** [comment]
Hi, good question. The model I used to fine-tune was trained on the same task with the much bigger SQuAD dataset. The idea was that it already "learned" to generate questions, albeit in a different language. So conceptually a similar idea to yours (Transfer Learning). More precisely, it could be described as a mixture of Cross-lingual learning and Sequential transfer learning.

That also kind of worked, since it peformed better than the plain t5-base, which was only pre-trained in the fashion descriped in the T5 paper. I also tried out a multilingual model, which was pre-trained on translations, including German (mbart-large-50),  but it performed slightly worse.

This are the results I obtained on the GermanQuAD test set:
| Model  | t5-small | t5-base | t5-base-pretrained | mbart-large-50 |
|--------|----------|---------|--------------------|----------------|
| BLEU-4 | 8.03     | 10.12   | 11.30              | 10.58          |

I'm planning to document all of that a bit better when I have the time. Training mT5 is an interesting idea, maybe even pre-training it on SQuAD too to see how that affects performance.

**Comment by TiloMichel** [comment]
Thanks for your answer and sharing your results, very interesting. I found one mT5 model on huggingface (https://huggingface.co/nbroad/mt5-small-qgen) but it isn't loaded on the frontend and no metrics were provided. Regarding the mbart-large-50 result, I would assume that if you fine-tune it additionally with the SQuAD dataset it would outperform the t5-base-pretrained, but that has to be checked I guess.
